homeassistant/components/homematic/sensor.py

```diff
@@ -45,54 +58,163 @@
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    "BRIGHTNESS": SensorEntityDescription(
+        key="BRIGHTNESS", native_unit_of_measurement="#", icon="mdi:invert-colors"
+    ),
 }
 
-HM_DEVICE_CLASS_HA_CAST = {
-    "HUMIDITY": DEVICE_CLASS_HUMIDITY,
-    "TEMPERATURE": DEVICE_CLASS_TEMPERATURE,
-    "ACTUAL_TEMPERATURE": DEVICE_CLASS_TEMPERATURE,
-    "LUX": DEVICE_CLASS_ILLUMINANCE,
-    "CURRENT_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "AVERAGE_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "LOWEST_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "HIGHEST_ILLUMINATION": DEVICE_CLASS_ILLUMINANCE,
-    "POWER": DEVICE_CLASS_POWER,
-    "CURRENT": DEVICE_CLASS_POWER,
-    "CONCENTRATION": DEVICE_CLASS_CO2,
+SENSOR_TYPES: dict[str, SensorEntityDescription | None] = {
+    "ACTUAL_TEMPERATURE": SENSOR_DESCRIPTIONS.get("TEMPERATURE_MEASUREMENT"),
+    "TEMPERATURE": SENSOR_DESCRIPTIONS.get("TEMPERATURE_MEASUREMENT"),
+    "LUX": SENSOR_DESCRIPTIONS.get("ILLUMINATION_MEASUREMENT"),
+    "CURRENT_ILLUMINATION": SENSOR_DESCRIPTIONS.get("ILLUMINATION_MEASUREMENT"),
+    "ILLUMINATION": SENSOR_DESCRIPTIONS.get("ILLUMINATION_MEASUREMENT"),
+    "AVERAGE_ILLUMINATION": SENSOR_DESCRIPTIONS.get("ILLUMINATION_TOTAL"),
+    "LOWEST_ILLUMINATION": SENSOR_DESCRIPTIONS.get("ILLUMINATION_TOTAL"),
+    "HIGHEST_ILLUMINATION": SENSOR_DESCRIPTIONS.get("ILLUMINATION_TOTAL"),
+    "POWER": SENSOR_DESCRIPTIONS.get("POWER_MEASUREMENT"),
+    "IEC_POWER": SENSOR_DESCRIPTIONS.get("POWER_MEASUREMENT"),
+    "CURRENT": SENSOR_DESCRIPTIONS.get("CURRENT_MEASUREMENT"),
+    "CONCENTRATION": SENSOR_DESCRIPTIONS.get("CONCENTRATION_MEASUREMENT"),
+    "ENERGY_COUNTER": SENSOR_DESCRIPTIONS.get("ENERGY_COUNTER_TOTAL_INCREASING"),
+    "IEC_ENERGY_COUNTER": SENSOR_DESCRIPTIONS.get("ENERGY_COUNTER_TOTAL_INCREASING"),
+    "VOLTAGE": SENSOR_DESCRIPTIONS.get("VOLTAGE_MEASUREMENT"),
+    "GAS_POWER": SENSOR_DESCRIPTIONS.get("GAS_POWER_MEASUREMENT"),
+    "GAS_ENERGY_COUNTER": SENSOR_DESCRIPTIONS.get(
+        "GAS_ENERGY_COUNTER_TOTAL_INCREASING"
+    ),
+    "RAIN_COUNTER": SENSOR_DESCRIPTIONS.get("RAIN_COUNTER"),
+    "WIND_SPEED": SENSOR_DESCRIPTIONS.get("WIND_SPEED"),
+    "WIND_DIRECTION": SENSOR_DESCRIPTIONS.get("WIND_DIRECTION"),
+    "WIND_DIRECTION_RANGE": SENSOR_DESCRIPTIONS.get("WIND_DIRECTION"),
+    "SUNSHINEDURATION": SENSOR_DESCRIPTIONS.get("SUNSHINEDURATION"),
+    "AIR_PRESSURE": SENSOR_DESCRIPTIONS.get("AIR_PRESSURE_MEASUREMENT"),
+    "FREQUENCY": SENSOR_DESCRIPTIONS.get("FREQUENCY"),
+    "VALUE": SENSOR_DESCRIPTIONS.get("VALUE"),
+    "VALVE_STATE": SENSOR_DESCRIPTIONS.get("VALVE_STATE"),
+    "CARRIER_SENSE_LEVEL": SENSOR_DESCRIPTIONS.get("CARRIER_SENSE_LEVEL"),
+    "DUTY_CYCLE_LEVEL": SENSOR_DESCRIPTIONS.get("DUTY_CYCLE_LEVEL"),
+    "BRIGHTNESS": SENSOR_DESCRIPTIONS.get("BRIGHTNESS"),
 }
 
-HM_ICON_HA_CAST = {"WIND_SPEED": "mdi:weather-windy", "BRIGHTNESS": "mdi:invert-colors"}
+
+def get_sensor_type_desc_attr(sensor_type, attr, default=None):
```

Comment:
Lets set the entity description property on the init and remove that property

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py

Reply:
Makes sense. Thanks 👍